### PR TITLE
Add definitions for W3C Web USB API

### PR DESCRIPTION
**Pull request by larsgk**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not provide its own types, and you can not add them.
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

File: types/w3c-web-usb/w3c-web-usb-tests.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
+// Modified from WebUSB spec: https://wicg.github.io/webusb/
+
+let connectedDevices: USBDevice[] = null;
```
Comment (kenchris) [inline types/w3c-web-usb/w3c-web-usb-tests.ts]: Why not

const connectedDevices: USBDevice[] = [] ?

File: types/w3c-web-usb/w3c-web-usb-tests.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+document.addEventListener('DOMContentLoaded', async () => {
+    const devices = await navigator.usb.getDevices();
+    devices.forEach(handleConnectedDevice);
+});
+
+let button = document.getElementById('request-device');
```
Comment (kenchris) [inline types/w3c-web-usb/w3c-web-usb-tests.ts]: you don't reassign, so why not const?

File: types/w3c-web-usb/w3c-web-usb-tests.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+button.addEventListener('click', async () => {
+    let device;
+    try {
+        device = await navigator.usb.requestDevice({ filters: [{
+            vendorId: 0xABCD,
+            classCode: 0xFF, // vendor-specific
+            protocolCode: 0x01
+        }]});
```
Comment (kenchris) [inline types/w3c-web-usb/w3c-web-usb-tests.ts]: why not add handleConnectedDevice here?

File: types/w3c-web-usb/w3c-web-usb-tests.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    if (i >= 0) {
+        connectedDevices.splice(i, 1);
+    }
+});
+
+async function handleConnectedDevice(device: USBDevice) {
+    connectedDevices.push(device);
+
+    await device.open();
+    if (device.configuration === null)
```
Comment (kenchris) [inline types/w3c-web-usb/w3c-web-usb-tests.ts]: {} around if, if spanning multiple lines

**Comment by kenchris** [comment]
Why didn't you tick all the boxes? you seem to have followed them

File: types/w3c-web-usb/w3c-web-usb-tests.ts (hunk 3 above)
Comment (larsgk) [reply types/w3c-web-usb/w3c-web-usb-tests.ts]: was making it close to the example code from spec

File: types/w3c-web-usb/w3c-web-usb-tests.ts (hunk 4 above)
Comment (larsgk) [reply types/w3c-web-usb/w3c-web-usb-tests.ts]: one line

**Pull request merged**